pkg/fixtures/workflows/github/helm/.github/workflows/azure-kubernetes-service-helm.yml

```diff
@@ -124,11 +128,15 @@ jobs:
           result=$(az aks command result --resource-group ${{ env.CLUSTER_RESOURCE_GROUP }} --name ${{ env.CLUSTER_NAME }} --command-id $command_id)
           echo "Helm upgrade result: $result"
           exitCode=$(az aks command result --resource-group ${{ env.CLUSTER_RESOURCE_GROUP }} --name ${{ env.CLUSTER_NAME }} --command-id $command_id --query exitCode -o tsv)
-
           if [ $exitCode -ne 0 ]; then
             exit $exitCode
           fi
 
       - name: Deploy application on public cluster
         if: steps.isPrivate.outputs.PRIVATE_CLUSTER != 'true'
-        run: helm upgrade --wait -i -f ${{ env.CHART_OVERRIDE_PATH }} --set ${{ env.CHART_OVERRIDES }} --set image.tag=${{ github.sha }} automated-deployment ${{ env.CHART_PATH }} --namespace ${{ env.NAMESPACE }} --create-namespace
+        run: |
+          if [ "${{ env.CLUSTER_RESOURCE_TYPE }}" == "Microsoft.ContainerService/fleets" ]; then
+            helm upgrade -i -f ${{ env.CHART_OVERRIDE_PATH }} --set ${{ env.CHART_OVERRIDES }} --set image.tag=${{ github.sha }} automated-deployment ${{ env.CHART_PATH }} --namespace ${{ env.NAMESPACE }}
```

Comment:
Are you absolutely sure the differences in this if branch are all needed?  Are they tested?

Helm documentation for `--create-namespace` says "if --install is set, create the release namespace **if not present**" -- on this basis, do we need to change it?